store/postgres/src/relational/index.rs

```diff
@@ -517,13 +563,137 @@ impl CreateIndex {
             ) => {
                 unique == o_unique
                     && method == o_method
-                    && columns == o_columns
+                    && Expr::is_same_kind_columns(columns, o_columns)
                     && cond == o_cond
                     && with == o_with
             }
         }
     }
 
+    pub fn is_id_immutable_table(&self, immutable: bool) -> bool {
+        // on imutable tables the id constraint is specified at table creation
+        if immutable {
+            match self {
+                CreateIndex::Unknown { .. } => (),
+                CreateIndex::Parsed { columns, .. } => {
+                    if columns.len() == 1 {
+                        if columns[0].is_id() {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        false
+    }
+
+    pub fn to_postpone(&self) -> bool {
+        fn has_prefix(s: &str, prefix: &str) -> bool {
+            s.starts_with(prefix)
+                || s.ends_with("\"") && s.starts_with(format!("\"{}", prefix).as_str())
+        }
+        match self {
+            CreateIndex::Unknown { .. } => false,
+            CreateIndex::Parsed {
+                name,
+                columns,
+                method,
+                ..
+            } => {
+                if *method != Method::BTree && *method != Method::Brin {
+                    return false;
+                }
+                if columns.len() == 1 && columns[0].is_id() {
+                    return false;
+                }
+                has_prefix(name, "attr_") && self.is_attribute_index()
+            }
+        }
+    }
+
+    pub fn name(&self) -> Option<String> {
+        match self {
+            CreateIndex::Unknown { .. } => None,
+            CreateIndex::Parsed { name, .. } => Some(name.clone()),
+        }
+    }
+
+    pub fn fields_exist_in_dest(&self, dest_table: &Table) -> bool {
+        fn column_exists(columns: &Vec<String>, column_name: &String) -> bool {
+            for c in columns.iter() {
+                if *c == *column_name {
+                    return true;
+                }
+            }
+            false
+        }
+
+        fn some_column_contained(expresion: &String, columns: &Vec<String>) -> bool {
+            for c in columns.iter() {
+                if expresion.contains(c) {
+                    return true;
+                }
+            }
+            false
+        }
+
+        let cols = &dest_table
+            .columns
+            .iter()
+            .map(|i| i.name.to_string())
+            .collect::<Vec<String>>();
+
+        match self {
+            CreateIndex::Unknown { defn: _ } => (),
```

Comment:
Is it correct that we return `true` for an index that we couldn't parse? It would be clearer if this arm had a `return` in it.

Reply:
Addressed it, thanks!

Reply:
Here I decided to err on the side of allowing more indexes to be created. Now in the context of with_nsp() returning error for this case I think we can keep it true (explicitly), so that the issue get's logged later on.